### DISCUSSION TITLE
Is there a list of validation errors from paserError and lexerError provider?

### DISCUSSION THREAD

**Discussion by durianwaffle** [eclipse-langium/langium#1715] [Q&A] [answered]
Hi Langium team,

The default validation errors from lexerError and paserError provider doesn't have localization. I want to override all the default validation error msgs because of localization need.

- I am wondering if I override paserError and lexerError provider, will the red squiggle lines disappear if there is paser error or lexer error? Or they will still be there?

- Is there a list of possible validation error msgs in paserError and lexerError provider so I can make sure I implemented all the checks in paserError and lexerError provider? Especially for the non-recoverable errors where AST node missed up.

**Comment by msujew** [comment, marked as answer]
Hey @durianwaffle,

for the parser errors, we already expose a `ParserErrorMessageProvider` in the `parser` part of the language services. You can override that to modify/translate the parser error messages.

For the lexer, you need to override the `Lexer` service yourself right now. I've submitted https://github.com/eclipse-langium/langium/pull/1716 to make this easier as well.

**Reply by durianwaffle** [reply]
Hi @msujew, thanks for providing the default lexer error message provider. Do you know how to import DefaultLexerErrorMessageProvider? I tried to import it from langiumParser, but it doesn't exist.
[langium/packages/langium/src/default-module.ts at main · eclipse-langium/langium](https://github.com/eclipse-langium/langium/blob/main/packages/langium/src/default-module.ts#L31)

**Reply by msujew** [reply]
You should be able to to just import it using `import { DefaultLexerErrorMessageProvider } from 'langium'`. Are you using the latest version of Langium? The class has only been added in release 3.3 of Langium.
